File: test/03_test.dart
import 'dart:io';
import 'package:test/test.dart';
import 'package:advent_of_code_2020/03.dart';

void main() {
  String contents = new File('./data/03.txt').readAsStringSync();
  
  String example = '..##.......\n#...#...#..\n.#....#..#.\n..#.#...#.#\n.#...##..#.\n..#.##.....\n.#.#.#....#\n.#........#\n#.##...#...\n#...##....#\n.#..#...#.#';

  group('A', () {
    test('Example 1', () {
      expect(A03(example),
          equals(7));
    });
    test('Solution', () {
      expect(A03(contents), equals(198));
    });
  });

  group('B', () {
    test('Example 1', () {
      expect(B03(example),
          equals(336));
    });
    test('Solution', () {
      expect(B03(contents), equals(5140884672));
    });
  });
}

File: test/03.test.js
const fs = require('fs');

const { A03, B03 } = require('../lib/03');
const input = fs.readFileSync('./data/03.txt', 'utf-8');

const example =
	'..##.......\n#...#...#..\n.#....#..#.\n..#.#...#.#\n.#...##..#.\n..#.##.....\n.#.#.#....#\n.#........#\n#.##...#...\n#...##....#\n.#..#...#.#';

describe('03', () => {
	describe('A', () => {
		it('Example 1', () => {
			expect(A03(example)).toBe(7);
		});

		it('Solution', () => {
			expect(A03(input)).toBe(198);
		});
	});

	describe('B', () => {
		it('Example 1', () => {
			expect(B03(example)).toBe(336);
		});

		it('Solution', () => {
			expect(B03(input)).toBe(5140884672);
		});
	});
});
